rfcs/rfc-001-smart-contracts-engine.md

```diff
@@ -37,154 +37,175 @@ version 0.1
-
-A client will create a transaction to create a new loader instance:
-
-`Solana_NewLoader(Loader Instance PubKey, proof of key ownership, space I need for my elf)`
-
-A client will then do a bunch of transactions to load its elf into the loader instance they created:
-
-`Loader_UploadElf(Loader Instance PubKey, proof of key ownership, pos start, pos end, data)`
-
-At this point the client can create a new instance of the module with its own instance address:
-
-`Loader_NewInstance(Loader Instance PubKey, proof of key ownership, Instance PubKey, proof of key ownership)`
-
-Once the instance has been created, the client may need to upload more user data to solana to configure this instance:
-
-`Instance_UploadModuleData(Instance PubKey, proof of key ownership, pos start, pos end, data)`
-
-Now clients can `start` the instance:
-
-`Instance_Start(Instance PubKey, proof of key ownership)`
-
 ## Runtime
 
-Our goal with the runtime is to have a general purpose execution environment that is highly parallelizable and doesn't require dynamic resource management. We want to execute as many contracts as we can in parallel, and have them pass or fail without a destructive state change.
-
-### State and Entry Point
-
-State is addressed by an account which is at the moment simply the PubKey.  Our goal is to eliminate dynamic memory allocation in the smart contract itself, so the contract is a function that takes a mapping of [(PubKey,State)] and returns [(PubKey, State')].  The output of keys is a subset of the input.  Three basic kinds of state exist:
+The goal with the runtime is to have a general purpose execution environment that is highly parallelizeable and doesn't require dynamic resource management. The goal is to execute as many contracts as possible in parallel, and have them pass or fail without a destructive state change.
 
-* Instance State
-* Participant State
-* Caller State
 
-There isn't any difference in how each is implemented, but conceptually Participant State is memory that is allocated for each participant in the contract.  Instance State is memory that is allocated for the contract itself, and Caller State is memory that the transactions caller has allocated.
+### State
 
+State is addressed by an account which is at the moment simply the PubKey.  Our goal is to eliminate dynamic memory allocation in the smart contract itself, so the contract is a function that takes a mapping of [(PubKey,State)] and returns [(PubKey, State')].
 
-### Call
-
+### Call Structure
 ```
-void call(
-    const struct instance_data *data,
-    const uint8_t kind[],  //instance|participant|caller|read|write
-    const uint8_t *keys[],
-    uint8_t *data[],
-    int num,
-    uint8_t dirty[],        //dirty memory bits
-    uint8_t *userdata,      //current transaction data
-);
+/// Call definition
+/// Signed portion
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct CallData {
+    /// number of keys to load, aka the to key
```

Comment:
I think you mean AKA and to quote the second "to"?  But if so, I don't see how "number of" and the "to" key might be the same thing, and how either of those might map to `Vec<PublicKey>` which should be `Vec<Pubkey>`.